internal/trie/node/header.go

```diff
@@ -7,62 +7,43 @@ import (
 	"io"
 )
 
+const (
+	leafHeader            byte = 1 // 01
+	branchHeader          byte = 2 // 10
+	branchWithValueHeader byte = 3 // 11
+)
+
 const (
 	keyLenOffset    = 0x3f
 	nodeHeaderShift = 6
 )
 
-// encodeHeader creates the encoded header for the branch.
-func (b *Branch) encodeHeader(writer io.Writer) (err error) {
+// encodeHeader writes the encoded header for the node.
+func encodeHeader(node *Node, writer io.Writer) (err error) {
 	var header byte
-	if b.Value == nil {
-		header = byte(BranchType) << nodeHeaderShift
-	} else {
-		header = byte(BranchWithValueType) << nodeHeaderShift
-	}
-
-	if len(b.Key) >= keyLenOffset {
-		header = header | keyLenOffset
-		_, err = writer.Write([]byte{header})
-		if err != nil {
-			return err
-		}
-
-		err = encodeKeyLength(len(b.Key), writer)
-		if err != nil {
-			return err
-		}
-	} else {
-		header = header | byte(len(b.Key))
-		_, err = writer.Write([]byte{header})
-		if err != nil {
-			return err
+	if node.Type() == Leaf {
+		header = leafHeader
+	} else { // branch
+		if node.Value == nil {
+			header = branchHeader
+		} else {
+			header = branchWithValueHeader
 		}
 	}
+	header <<= nodeHeaderShift
 
-	return nil
-}
-
-// encodeHeader creates the encoded header for the leaf.
-func (l *Leaf) encodeHeader(writer io.Writer) (err error) {
-	header := byte(LeafType) << nodeHeaderShift
-
-	if len(l.Key) < 63 {
-		header |= byte(len(l.Key))
+	if len(node.Key) < keyLenOffset {
+		header |= byte(len(node.Key))
 		_, err = writer.Write([]byte{header})
 		return err
 	}
 
-	header |= keyLenOffset
+	header = header | keyLenOffset
```

Comment:
In many places we're using short self-assigned vars I think would be nice to keep here as well
```suggestion
	header |= keyLenOffset
```

Reply:
Indeed, although most of these get thrown out in #2530 ([that particular line gets removed and I do use the short `|` form too!](https://github.com/ChainSafe/gossamer/pull/2530/files#diff-63521964d6b19c9f06c40c3276755b2070ec87f12744cbea72d74ca32f152561L41)), so I'll not add more diffs to have an easier rebase for the next PR which contains those changes :wink: